web/src/main/java/org/wso2/testgrid/web/sso/SSOContextEventListener.java

```diff
@@ -0,0 +1,82 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.testgrid.web.sso;
+
+import org.wso2.carbon.identity.sso.agent.SSOAgentConstants;
+import org.wso2.carbon.identity.sso.agent.SSOAgentException;
+import org.wso2.carbon.identity.sso.agent.bean.SSOAgentConfig;
+import org.wso2.carbon.identity.sso.agent.saml.SSOAgentX509Credential;
+import org.wso2.carbon.identity.sso.agent.saml.SSOAgentX509KeyStoreCredential;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+
+/**
+ * Implementation of context event listener for fetch SSO related details.
+ */
+public class SSOContextEventListener implements ServletContextListener {
+
+    private static Logger logger = Logger.getLogger(SSOContextEventListener.class.getName());
+
+    private static Properties properties = new Properties();
+
+    /**
+     * Fetch relevant details from the property files and JKS file.
+     */
+    public void contextInitialized(ServletContextEvent servletContextEvent) {
+
+        try {
+            if (servletContextEvent.getServletContext().getContextPath().contains("testgrid")) {
+                properties.load(servletContextEvent.getServletContext().
+                        getResourceAsStream("/WEB-INF/classes/testgrid-sso.properties"));
+            }
+            InputStream keyStoreInputStream = servletContextEvent.getServletContext().
+                    getResourceAsStream("/WEB-INF/classes/wso2carbon.jks");
+            SSOAgentX509Credential credential =
+                    new SSOAgentX509KeyStoreCredential(keyStoreInputStream,
+                            properties.getProperty("KeyStorePassword").toCharArray(),
+                            properties.getProperty("IdPPublicCertAlias"),
+                            properties.getProperty("PrivateKeyAlias"),
+                            properties.getProperty("PrivateKeyPassword").toCharArray());
+            SSOAgentConfig config = new SSOAgentConfig();
+            config.initConfig(properties);
+            config.getSAML2().setSSOAgentX509Credential(credential);
+            servletContextEvent.getServletContext().
+                    setAttribute(SSOAgentConstants.CONFIG_BEAN_NAME, config);
+        } catch (IOException | SSOAgentException e) {
+            logger.log(Level.SEVERE, e.getMessage(), e);
+        }
+    }
+
+    public void contextDestroyed(ServletContextEvent servletContextEvent) {
+
+    }
+
+    /**
+     * Get the properties of the sample
+     * @return Properties
+     */
+    public static Properties getProperties() {
```

Comment:
I do not think it is correct to export properties like this. As I see, these properties contains the testgrid-sso.properties file's contents. We should provide a config provider for that purpose.

Reply:
For SSOAgentConfig's initConfig method, we needs to pass these properties as java.util.Properties. Therefore we will be needing to represent them as this.

Reply:
Yes, the requirement is clear. But this is not the way to do it. Please re-think the design.

Reply:
Created a separate `SSOConfigurationReader` class and moved these external readings into that.